src/components/rating/subcomponents/Star.jsx

```diff
@@ -3,19 +3,21 @@ import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import Icon, {TYPE, ICON_COLOR} from '../../icons/Icon';
 
-const Star = ({size, checked, onClick}) => {
+const Star = ({size, checked, onClick, index, ...props}) => {
   const starClass = classnames('sg-rate-box__star', {
     'sg-rate-box__star--checked': checked
   });
+  const onClickStar = () => onClick(index);
```

Comment:
you are passing index only to pass it back to the parent - it'd be better if parent did the `index` binding to the onClick function

Reply:
but index binding creating new function everytime

Reply:
ok, but you can still create a new function in parent and avoid passing `index` here